### PR TITLE
Updated main configuration guide to Flex

### PR DESCRIPTION
**Pull request by wouterj**

### CODE REVIEW

File: configuration.rst [hunk 1]
```diff
@@ -304,93 +214,18 @@ There is also a ``.env`` file which is loaded. Its contents become environment v
...
+The ``.env`` file is special, because it defines the values that usually change
+on each server. For example, the database credentials on your local development
+machine might be different from your workmates. That's why this file is **not
+committed to the shared repository** and is only stored on your machine. In
+fact, the ``.gitignore`` file that comes with Symfony prevents it from being
+committed.
+
+However, a ``.env.dist`` file *is* committed (with dummy values). This file
+isn't read by Symfony: it's just a reference so that Symfony knows which
+parameters need to be defined in the ``.env.dist`` file. If you add or remove
```
Comment (javiereguiluz) [inline configuration.rst]: `need to be defined in the ``.env.dist`` file` -> `need to be defined in the ``.env`` file` ?

File: configuration.rst [hunk 2]
```diff
@@ -6,92 +6,73 @@ Configuring Symfony (and Environments)
...
-via configuration files that live - by default - in the ``app/config`` directory.
+via configuration files that live - by default - in the ``config/`` directory.
 
-Configuration: config.yml
--------------------------
+Configuration: config/packages/
+-------------------------------
 
-The main configuration file is called ``config.yml``:
+The configuration for a package can be found in ``config/packages``. For
```
Comment (javiereguiluz) [inline configuration.rst]: `can be found in config/packages` --> `can be found in config/packages/`  please always add a trailing slash when referencing directories so the reader never wonders if it's a dir or a file without extension. Thanks!

**Review by javiereguiluz — APPROVED** [review]
Nice changes here! Thanks Wouter.

File: configuration.rst [hunk 3]
```diff
@@ -6,92 +6,73 @@ Configuring Symfony (and Environments)
...
-via configuration files that live - by default - in the ``app/config`` directory.
+via configuration files that live - by default - in the ``config/`` directory.
 
-Configuration: config.yml
--------------------------
+Configuration: config/packages/
+-------------------------------
 
-The main configuration file is called ``config.yml``:
+The configuration for a package can be found in ``config/packages/``. For
```
Comment (weaverryan) [inline configuration.rst]: > for each package...

File: configuration.rst [hunk 4]
```diff
@@ -111,143 +90,83 @@ Configuration Reference & Dumping
...
+
+The parameters Key: Parameters (Variables)
+------------------------------------------
 
-Symfony's main configuration file is ``app/config/config.yml``. But, for organization,
-it *also* loads other configuration files via its ``imports`` key:
+The configuration has some special top-level keys. One of them is called
+``parameters``: it's used to define *variables* that can be referenced in *any*
+other configuration file. For example, when installing the *translation*
+package, ``config/services.yaml`` defines a ``locale`` parameter:
```
Comment (weaverryan) [inline configuration.rst]: How about:

> For example, when you install the *translation* package, a `locale` parameter is added to `config/services.yaml`:

File: configuration.rst (hunk 1 above)
Comment (weaverryan) [inline configuration.rst]: `parameters` -> `variables` (to make it a bit clear that these aren't parameters)

File: configuration.rst [hunk 5]
```diff
@@ -304,93 +214,18 @@ There is also a ``.env`` file which is loaded. Its contents become environment v
...
+on each server. For example, the database credentials on your local development
+machine might be different from your workmates. That's why this file is **not
+committed to the shared repository** and is only stored on your machine. In
+fact, the ``.gitignore`` file that comes with Symfony prevents it from being
+committed.
+
+However, a ``.env.dist`` file *is* committed (with dummy values). This file
+isn't read by Symfony: it's just a reference so that Symfony knows which
+parameters need to be defined in the ``.env.dist`` file. If you add or remove
+keys to ``.env``, add or remove them from ``.env`` too, so both files are
```
Comment (weaverryan) [inline configuration.rst]: the second should be `.env.dist`.

**Review by weaverryan — CHANGES_REQUESTED** [review]

File: configuration.rst [hunk 6]
```diff
@@ -111,143 +90,83 @@ Configuration Reference & Dumping
...
-
-The parameters Key: Parameters (Variables)
-------------------------------------------
-
-Another special key is called ``parameters``: it's used to define *variables* that
-can be referenced in *any* other configuration file. For example, in ``config.yml``,
-a ``locale`` parameter is defined and then referenced below under the ``framework``
-key:
+This parameter is then referenced in the framework config in
+``config/packages/translation.yaml``:
```
Comment (yceruto) [inline configuration.rst]: ``config/packages/framework.yaml`` same for xml and php versions.

**Comment by wouterj** [comment]
Ready to merge!

**Comment by weaverryan** [comment]
Wooo! Thank you for this BIG pr Wouter!

**Pull request closed without merging**